torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -2008,105 +1989,83 @@ RegisterOperators reg2({
-          "aten::mul_." decl_type "(" decl_type                                     \
-          "[](a!) l, int n) -> " decl_type "[](a!)",                                \
-          listMulIntLeftInPlace<value_type>,                                        \
-          aliasAnalysisFromSchema())
+    // these ops are generic over the list element type.
+    // CREATING GENERIC_LIST_OPS
+    Operator(
+        "aten::select.t(t[](a) list, int idx) -> t(*)",
+        listSelect,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::__getitem__.t(t[](a) list, int idx) -> t(*)",
+        listSelect,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::append.t(t[](a!) self, t(c -> *) el) -> t[](a!)",
+        listAppend,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::reverse.t(t[](a!) self) -> ()",
+        listReverse,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::extend.t(t[](a!) self, t[] other) -> ()",
+        listExtend,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::copy.t(t[](a) self) -> t[]",
+        listCopy,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::_set_item.t(t [](a!) l, int idx, t(b -> *) el) -> t[](a!)",
+        listSetItem,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::clear.t(t[](a!) self) -> ()",
+        listClear,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::Delete.t(t[](a!) self, int idx) -> ()",
+        listDelete,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::insert.t(t[](a!) self, int idx, t(b -> *) el) -> ()",
+        listInsert,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::pop.t(t[](a!) self, int idx=-1) -> t(*)",
+        listPop,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::add.t(t[] a, t[] b) -> t[]",
+        listAdd,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::add_.t(t[](a!) self, t[] b) -> t[]",
+        listInplaceAdd,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::slice.t(t[] l, int start, int end=9223372036854775807, int step=1) -> t[]",
+        listSlice,
+        aliasAnalysisFromSchema()),
+    Operator("aten::list.t(t[] l) -> t[]", listList, aliasAnalysisFromSchema()),
+    Operator(
+        "aten::mul.left_t(t[] l, int n) -> t[]",
+        listMulIntLeft,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::mul.right_(int n, t[] l) -> t[]",
+        listMulIntRight,
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::mul_.t(t[](a!) l, int n) -> t[](a!)",
+        listMulIntLeftInPlace,
+        aliasAnalysisFromSchema()),
 
-    CREATE_GENERIC_LIST_OPS("t", IValue),
+    // registered as Any[] so that heterogenous tuples can be called with len()
+    Operator(
+        "aten::len.t(Any[] a) -> int",
+        listLen<IValue>,
```

Comment:
The template parameter on `listLen` can also be deleted

Reply:
that would be a BC incompat change for mobile

Reply:
Why is that? it looks like `listLen` isn't used outside of this file

Reply:
oop yea you're right.